### PR TITLE
Inherit directly from the BasePlugin for autotools

### PR DESCRIPTION
**Pull request by sergiusens**
Signed-off-by: Sergio Schvezov <sergio.schvezov@canonical.com>

### CODE REVIEW

File: snapcraft/plugins/autotools.py [hunk 1]
```diff
@@ -74,4 +75,7 @@ def build(self):
                 self.run(['autoreconf', '-i'], cwd=self.builddir)
         self.run(['./configure', '--prefix='] +
                  self.options.configflags)
-        self.run(['make', 'install', 'DESTDIR=' + self.installdir])
+        self.run_output(['ls'],
+                        cwd=self.builddir)
```
Comment (ted-gould) [inline snapcraft/plugins/autotools.py]: I think that the 'ls' is a leftover here.

Comment (elopio) [inline snapcraft/plugins/autotools.py]: It seems that self.builddir is the default when cwd is not passed.

**Comment by elopio** [comment]
I confirm that with this change the libpipeline example builds in xenial.
I don't understand the run_output method, it needs a docstring.

File: snapcraft/plugins/autotools.py [hunk 2]
```diff
@@ -74,4 +75,7 @@ def build(self):
...
+        self.run(['make', 'install', 'DESTDIR=' + self.installdir],
+                 cwd=self.builddir)
```
Comment (elopio) [inline snapcraft/plugins/autotools.py]: It seems that self.builddir is the default when cwd is not passed.

**Comment by sergiusens** [comment]
all fixed

**Comment by elopio** [comment]
:+1: looks better, and the test still passes.

**Comment by ted-gould** [comment]
+1

**Pull request merged**